### PR TITLE
chore(api_map): remove unused api and repalce error name api

### LINKED ISSUE #5343: v9.0.0 `LV_STYLE_TRANSFORM_SCALE` not defined
### LVGL version

V9.0.0

### What happened?

In `lv_api_map.h`: line 246:

`#define LV_STYLE_TRANSFORM_ZOOM       LV_STYLE_TRANSFORM_SCALE`

`LV_STYLE_TRANSFORM_SCALE` is not defined, there must be a typo somewhere.

https://github.com/lvgl/lvgl/blob/fb0fc86895331bac9f5ce514ef6324342e8ed3c6/src/lv_api_map.h#L246C1-L246C63

### How to reproduce?

Compilation issue found by Berry mapping

### PR DESCRIPTION
**Pull request by lhdjply**
Help us review this PR! Anyone can [approve it or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### Description of the feature or fix

fix #5343 
chore(api_map): remove unused api and repalce error name api

### Checkpoints
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)

### CODE REVIEW

**Review by C47D — APPROVED** [review]

**Review by FASTSHIFT — DISMISSED** [review]

**Review by PGNetHun — APPROVED** [review]

File: src/lv_api_map.h [hunk 1]
```diff
@@ -137,7 +136,7 @@ static inline void lv_obj_move_background(lv_obj_t * obj)
 #define lv_disp_get_layer_top            lv_display_get_layer_top
 #define lv_disp_get_layer_sys            lv_display_get_layer_sys
 #define lv_disp_get_layer_bottom         lv_display_get_layer_bottom
-#define lv_disp_add_event                lv_display_add_event
+#define lv_disp_add_event                lv_display_add_event_cb
```
Comment (kisvegabor) [inline src/lv_api_map.h]: Display events were introduced in v9, so no v8 mapping is required. Please remove `lv_disp_add_event`, `lv_disp_get_event_count`, `lv_disp_get_event_dsc`, and `lv_disp_remove_event`

**Comment by kisvegabor** [comment]
In [b81d682](https://github.com/lvgl/lvgl/pull/5367/commits/b81d682888a7673c35feb86332801f5ac6b61c04)  all `lv_disp_` was removed. [Here](https://github.com/lvgl/lvgl/pull/5367#discussion_r1457162072) I meant only the 4 mentioned event related ones.

**Comment by lhdjply** [comment]
> In [b81d682](https://github.com/lvgl/lvgl/pull/5367/commits/b81d682888a7673c35feb86332801f5ac6b61c04) all `lv_disp_` was removed. [Here](https://github.com/lvgl/lvgl/pull/5367#discussion_r1457162072) I meant only the 4 mentioned event related ones.

The functions I'm removing now are not in 8.3, not just those four.

**Comment by lhdjply** [comment]
> In [b81d682](https://github.com/lvgl/lvgl/pull/5367/commits/b81d682888a7673c35feb86332801f5ac6b61c04) all `lv_disp_` was removed. [Here](https://github.com/lvgl/lvgl/pull/5367#discussion_r1457162072) I meant only the 4 mentioned event related ones.

The functions I'm removing now are not in 8.3, not just those four.

For example, `lv_disp_create` don't have in v8.3
Screenshot_20240118_190159_com huawei browser

**Comment by kisvegabor** [comment]
Oh, you are right. I've just resolved a conflict. When the CI passes I mereg it.

**Review by kisvegabor — APPROVED** [review]

**Comment by lhdjply** [comment]
CI is passed.But it is not auto merge. @kisvegabor

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**